src/pages/index.js

```diff
@@ -128,24 +188,75 @@ function closeModal(modal) {
 
 function handleEditFormSubmit(evt) {
   evt.preventDefault();
-  profileName.textContent = editModalNameInput.value;
-  profileDescription.textContent = editModalDescriptionInput.value;
-  closeModal(editModal);
+  const submitButton = evt.submitter;
+  setButtonText(submitButton, true);
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitButton, false);
+    });
 }
 function handleAddCardSubmit(evt) {
   evt.preventDefault();
 
-  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
-  const cardElement = getCardElement(inputValues);
+  const submitButton = evt.submitter;
+  setButtonText(submitButton, true);
 
-  cardsList.prepend(cardElement);
-  evt.target.reset();
-  const inputList = Array.from(
-    cardForm.querySelectorAll(settings.inputSelector)
-  );
-  resetValidation(cardForm, inputList, settings);
-  closeModal(cardModal);
+  const inputValues = {
+    name: cardNameInput.value,
+    link: cardLinkInput.value,
+  };
+
+  api
+    .addCard(inputValues)
+    .then((newCard) => {
+      const cardElement = getCardElement(newCard);
+      cardsList.prepend(cardElement);
+      closeModal(cardModal);
+      evt.target.reset();
+      disableButton(submitButton, settings);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      setButtonText(submitButton, false);
+    });
 }
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+
+  const submitButton = evt.submitter;
+  setButtonText(submitButton, true);
+  api
+    .editAvatarInfo({ avatar: avatarInput.value })
+    .then((data) => {
+      profileAvatar.src = data.avatar;
+      profileAvatar.alt = "Profile avatar";
+      evt.target.reset();
+      disableButton(submitButton, settings);
+      const inputList = Array.from(
+        avatarForm.querySelectorAll(settings.inputSelector)
+      );
+      resetValidation(avatarForm, inputList, settings);
```

Comment:
`      resetValidation(avatarForm, inputList, settings);
`

Please, delete it

`resetValidation` is for clearing possible validation errors when you open the **profile** form, because you insert **valid** data into its inputs. But when you add a new card or change the `avatar`, you don't have any errors after submission.

 It’s a **waste of resources**.